readme.md

```diff
@@ -312,17 +317,14 @@ For a TS example see the following [discussion](https://github.com/pmndrs/zustan
 import { State, StateCreator } from 'zustand'
 import produce, { Draft } from 'immer'
 
-// Immer V8 or lower
-const immer = <T extends State>(
-  config: StateCreator<T, (fn: (draft: Draft<T>) => void) => void>
-): StateCreator<T> => (set, get, api) =>
-  config((fn) => set(produce(fn) as (state: T) => T), get, api)
-
-// Immer V9
-const immer = <T extends State>(
-  config: StateCreator<T, (fn: (draft: Draft<T>) => void) => void>
-): StateCreator<T> => (set, get, api) =>
-  config((fn) => set(produce<T>(fn)), get, api)
+const immer = <T extends State>config: StateCreator<T>): StateCreator<T> => 
+  (set, get, api) => config((partial, replace) => {
+    const nextState =
+      typeof partial === 'function'
+        ? produce(partial as (state: T) => T)
+        : partial as Draft<T>
```

Comment:
Isn't this type just `T` ?
Hm, `config` doesn't seem to be typed properly.
Would you try it with codesandbox? We should still want to have both for v8 and v9?

Reply:
Yea, you are right again :) I created a sandbox for this:

https://codesandbox.io/s/laughing-nash-vz53e?file=/src/zustand/middlewares/immer.ts

When I switch from immer v8 to immer v9 I see no difference. The typings should apply for both versions.